msu/utilities/log.nut

```diff
@@ -0,0 +1,87 @@
+::MSU.Log <- {
+	// maxLen is the maximum length of an array/table whose elements will be displayed
+	// maxDepth is the maximum depth at which arrays/tables elements will be displayed
+	// advanced allows the ID of the object to be displayed to identify different/identical objects
+	function printStackTrace( _maxDepth = 0, _maxLen = 10, _advanced = false )
+	{
+		local count = 2;
+		local string = "";
+		while (getstackinfos(count) != null)
+		{
+			local line = getstackinfos(count++);
+			string += "Function:\t\t";
+
+			if (line.func != "unknown")
+			{
+				string += line.func + " ";
+			}
+
+			string += "-> " + line.src + " : " + line.line + "\nVariables:\t\t";
+
+			foreach (key, value in line.locals)
+			{
+				string += this.getLocalString(key, value, _maxLen, _maxDepth, _advanced);
+			}
+			string = string.slice(0, string.len() - 2);
+			string += "\n";
+		}
+		::logInfo(string);
+	}
+
+	function printData( _data, _maxDepth = 1, _advanced = false )
```

Comment:
we should have a separate getData function which doesn't print, also maybe this should accept a maxLen parameter and print to that if maxLen > _data.len()

Reply:
Agree about maxLen, but why do we need the getData function in ::MSU.Log?